dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/TypeComparators.java

```
@@ -0,0 +1,28 @@
+package datadog.trace.agent.tooling.bytebuddy.matcher;
+
+import java.util.Comparator;
+import net.bytebuddy.description.type.TypeDefinition;
+import net.bytebuddy.description.type.TypeDescription;
+
+public class TypeComparators {
+
+  public static final Comparator<TypeDescription> TYPE_DESCRIPTION =
+      new TypeDescriptionComparator();
+  public static final Comparator<TypeDefinition> TYPE_DEFINITION = new TypeDefinitionComparator();
+
+  private static class TypeDescriptionComparator implements Comparator<TypeDescription> {
+
+    @Override
+    public int compare(TypeDescription o1, TypeDescription o2) {
+      return o1.getSimpleName().compareTo(o2.getSimpleName());
```

Is it ok not to do `null` check?

---

These are bytebuddy types. As far as I'm aware, the library will throw exceptions rather than produce null values.

---

So I'm wondering if this has something to do with the Netty failure. From digging into ByteBuddy a bit it seems like the default `equals` for those types work on `getName` and not `getSimpleName`